docs/aws_sso.md

```diff
@@ -0,0 +1,56 @@
+If you are working with AWS, there are two things you should avoid at all costs:
+
+* Use account’s root credentials for day-to-day work
+* Provision individual users in your account and generate credentials for them
+
+AWS Single Sign On (SSO) with Athenz allows you to avoid both. Because an Athenz
+domain is directly related to an AWS account, the same authorization mechanism
+can be leveraged for SSO to the AWS console, via SAML-based integration. The
+Identity Provider service validates the user and then uses Athenz
+to determine the AWS role(s) the user can assume. This set of roles also indirectly
+identifies all the accounts the user can access from the console. In its simplest
+form, the single (potentially large) list can be presented to AWS in a SAML
+document, and AWS allows selection of the account/role to use.
+
+## Onboard to AWS
+
+To register an AWS Account with a domain, run the following command:
+```
+zms-cli -d <domain-name> set-aws-account <aws-account-id>
+```
+
+In the Athenz UI, when you select your domain,
+you should see your onboarded AWS account number.
+
+Once your AWS account creation has successfully completed, it should create the
+following 3 required roles for AWS Console SSO with their corresponding policies:
+
+* aws.fed.admin.user - Members of this role get admin privileges for your AWS
+account. This is ROOT for AWS accounts. No one should be assigned this role,
+unless they are performing ROOT functions
+* aws.fed.power.user - This role restricts access to IAM functions, VPC core
+functions, etc.  Your day to day tasks should use this role.
+* aws.fed.read.only - ReadOnly Access to the AWS account.
+
+![AWS Federated roles in Athenz](images/aws_fed_roles.png)
+
+## Add members to AWS SSO roles
+
+In the Athenz UI,
+
+Follow these steps to add any users in one of those 3 aws sso roles. Since
+SAML integration does not carry out standard Athenz Authorization checks
+but rather reads the contents of the role to find out what IAM roles the
+user has access, the users must be explicitly added to one of these roles
+to get login access to the configured AWS account.
+
+View a particular role’s details by clicking on the Members icon
+in the role name line. The server shows all the members plus the audit history
+of when and who added or deleted any of the members in that role.
+Once the role members view is displayed, you can add any new members to
+the role by entering their identity name. To add new members enter their ids and
```

Comment:
Removed reference to Okta. Original text:
"...the role by entering their identity name. The users are identified as
 follows: `user.<okta-short-id>`. To add new members enter their ids and..."